### DISCUSSION TITLE
Search: Use "and" in query in order to apply all keywords?

### DISCUSSION THREAD

**Discussion by alexisLefebvre** [photoprism/photoprism#874] [Q&A] [answered]
As [noted by an user](https://github.com/photoprism/photoprism/issues/747#issuecomment-757156618), PhotoPrism use an `OR` query so that if we search with several keywords, it will show photos that match that at least one keyword, not the photos that match all the keywords.

That's also explained in the [documentation](https://docs.photoprism.org/user-guide/organize/search/#search):

> A search for `cat blue berlin` will find all photos that either display a cat, have blue as main color or have been taken in Berlin.

Let's say that we want to see only photos of blue cat. Is it possible? How can we do this?

---

Example from demo, let's say that someone called Tim Platrow exist and that we try to find him: https://demo.photoprism.org/photos?view=cards&order=added&q=tim%20paltrow&public=true&quality=3

> ![image](https://user-images.githubusercontent.com/2071331/104137948-f0843100-53a0-11eb-94cf-cc1d27aa3197.png)

Is it possible to ignore the images with Tim Robbins and Gwyneth Paltrow? (it should return no result in this case)

**Comment by lastzero** [comment, marked as answer]
https://demo.photoprism.org/photos?view=cards&order=added&q=cat%20color%3Agreen%20country%3Aza&public=true&quality=3

**Reply by alexisLefebvre** [reply]
Thanks, I'm sorry I missed that part of the documentation: https://docs.photoprism.org/user-guide/organize/search/#and-search

It looks like we can't use this feature to search a combination of several labels: https://demo.photoprism.org/photos?view=cards&order=added&q=label%3Adog%20label%3Acat&public=true&quality=3 show only cats and https://demo.photoprism.org/photos?view=cards&order=added&q=label%3Acat%20label%3Adog&public=true&quality=3 show only dogs

**Reply by lastzero** [reply]
If this is important to you, please open a feature request 👍 

My personal impression is that most people don't use the search at all and just keep on scrolling, so I spent my time improving this use case first.

**Reply by alexisLefebvre** [reply]
Is there a query parameter to search one (or several) keywords?

Let's say I want to find that picture, see the keyword in bottom right:

> ![image](https://user-images.githubusercontent.com/2071331/104138268-45c14200-53a3-11eb-99e1-6b5d87155969.png)

I tried with `label:` but it returns an error: https://demo.photoprism.org/photos?view=cards&order=added&q=label%3Asanctuary&public=true&quality=3

**Reply by lastzero** [reply]
For the difference between labels and keywords, see https://docs.photoprism.org/user-guide/faq/#whats-the-difference-between-keywords-and-labels

**Reply by lastzero** [reply]
So when you create a label with that keywords, this might work (probably requires re-indexing). Without a scope like `label:`, PhotoPrism will search for keywords unless a label with the exact name exists.

**Reply by alexisLefebvre** [reply]
Thanks for the explanations, I created #881 and #882.

**Comment by lastzero** [comment]
Searching by location is somewhat difficult in that there are different levels like street, city, country, and continent as well as different names for the same place depending on the language. We'll eventually master that too, but not today.

**Reply by alexisLefebvre** [reply]
That was a terrible example, sorry. I updated the question to focus on the labels.

**Comment by graciousgrey** [comment]
AND search and keywords search filter are available now. https://docs.photoprism.org/user-guide/organize/search/

**Reply by streiman** [reply]
AND works very well, but now I found that a NOT operator would be great like

"car & NOT bmw"

to get all pictures with keyword car, but not with the keyword bmw. Is this already anywhere discussed or even possible?

**Reply by graciousgrey** [reply]
A NOT operator would be definitely nice to have! I guess we need to limit it to certain search filters, so that it does not affect search performance that much. Feel free to open a new issue for this :)
